replica/table.cc

```diff
@@ -3057,6 +3062,10 @@ future<> table::flush(std::optional<db::replay_position> pos) {
     if (pos && *pos < _flush_rp) {
         co_return;
     }
+    // There is nothing to flush if the table was stopped.
+    if (_pending_flushes_phaser.is_closed()) {
```

Comment:
I am not quite getting this. Was not the intended effect earlier to _wait_ for any currently active flush to finish before we return here? Is the idea here that we've forced all flushes to disk through the call to `_sg_manager->stop_storage_groups();`? (The logic flow from there is somewhat indirect)?

Reply:
Yes, the change in table::stop to move the call to `_pending_flushes_phaser.close()` after `_sg_manager->stop_storage_groups()` means that is the phaser is_closed here, it means that all memtables were flushed via `stop_storage_groups` so there's nothing to wait for and we can just co_return immediately.